src/components/pages/not-found.jsx

```diff
@@ -1,31 +1,74 @@
-import {Link, locationShape} from "react-router"
+import { Link, locationShape, routerShape } from "react-router"
 import React, {PropTypes} from "react"
 import DocumentTitle from "react-document-title"
-
+import {FormattedMessage} from "react-intl"
 
 const NotFoundPage = React.createClass({
+  contextTypes: {
+    router: routerShape.isRequired
+  },
+
   propTypes: {
     location: locationShape.isRequired,
     message: PropTypes.string,
+    countryPackageName: PropTypes.string
+  },
+
+  handleInputChange(event) {
+    this.setState({inputValue: event.target.value})
   },
+
+  handleSubmit(event) {
+    event.preventDefault()
+    this.context.router.push({
+      query: {q: this.state.inputValue},
+      hash: `#search-input`,
+    })
+  },
+
   render() {
     const {pathname} = this.props.location
-    const message = this.props.message || `La page « ${pathname} » n'existe pas.`
+    const countryPackageName = this.props.countryPackageName
+    const changelogURL = `http://www.github.com/openfisca/${countryPackageName}/blob/master/CHANGELOG.md`
     return (
       <DocumentTitle title={`Page non trouvée - Explorateur de la législation`}>
         <div>
           <div className="page-header">
-            <h1>Page non trouvée</h1>
+            <h1>
+            <FormattedMessage id = "pageDoesNotExist" values=
+              {{inputValueRef:`${pathname}`}}/>
+            </h1>
           </div>
-          <div className="alert alert-danger">
-            {message}
+          <div className="alert alert-info">
+            <p> <FormattedMessage id = "checkChangelog" values=
+              {{changelogURLLink:
+                <a href = {changelogURL} target = "_blank">changelog</a>
+              }}/>
+            </p>
+          </div>
+          <Link className="btn btn-default" to="/"><FormattedMessage id = "backToHP"/></Link>
+          <hr></hr>
+          <h3><FormattedMessage id = "explore"/></h3>
+          <div>
+            <form onSubmit={this.handleSubmit}>
```

Comment:
This code is really similar to the one in `home.jsx`. It would be relevant to create a `SearchBar` component, and include it in both pages.